src/model/quotes.js

```diff
@@ -168,8 +168,22 @@ class QuotesModel {
       throw ErrorHandler.CreateInternalServerFSPIOPError('Missing quoteRequest', null, fspiopSource)
     }
 
-    await this.db.getParticipant(fspiopSource, LOCAL_ENUM.PAYER_DFSP, quoteRequest.amount.currency, ENUM.Accounts.LedgerAccountType.POSITION)
+    // In fspiop api spec 2.0, to support FX, `supportedCurrencies` can be optionally passed in via the payer property.
+    // If `supportedCurrencies` is present, then payer FSP must have position accounts for all those currencies.
+    // If it is not passed in, then we validate against the `amount` currency only.
+    const payerFspCurrencies = []
+    if (quoteRequest.payer?.supportedCurrencies) {
+      payerFspCurrencies.push(...quoteRequest.payer.supportedCurrencies)
+    }
+    payerFspCurrencies.push(quoteRequest.amount.currency)
```

Comment:
I think this statement should be in else. We need to validate against the currency in the amount object only if there is no supportedCurrencies set.